choreography/src/choreography/Fabrikam.Choreography.ChoreographyService/Services/EventRepository.cs

```diff
@@ -3,48 +3,42 @@
 using System.Linq;
 using System.Net.Http;
 using System.Threading.Tasks;
-using Microsoft.Azure.EventGrid;
-using Microsoft.Azure.EventGrid.Models;
+using Azure;
+using Azure.Messaging.EventGrid;
 using Microsoft.Extensions.Logging;
 
 namespace Fabrikam.Choreography.ChoreographyService.Services
 {
     public class EventRepository : IEventRepository
     {
-        private readonly TopicCredentials topicCredentials;
-        private readonly EventGridClient eventGridClient;
-        private readonly string eventGridHost;
-        private readonly string[] Topics;
+        private readonly AzureKeyCredential topicCredentials;
+        private readonly EventGridPublisherClient eventGridClient;
+        private readonly string[] topics;
         private readonly Random random;
 
-        public EventRepository(string eventGridHost,string eventKey, string Topics)
+        public EventRepository(string eventGridHost,string eventKey, string topics)
         {       
-            topicCredentials = new TopicCredentials(eventKey);
-            eventGridClient = new EventGridClient(topicCredentials);
-            this.eventGridHost = eventGridHost;
-            this.Topics = Topics.Split(",");
+            topicCredentials = new AzureKeyCredential(eventKey);
+            eventGridClient = new EventGridPublisherClient(new Uri(eventGridHost), topicCredentials);
+            this.topics = topics.Split(",");
             random = new Random();
         }
 
         public string GetTopic()
         {
-            return Topics[random.Next(0, Topics.Length)];
+            return topics[random.Next(0, topics.Length)];
         }
 
         public async Task SendEventAsync(List<EventGridEvent> listEvents)
         {
 
             try
-            {             
-                var response = await eventGridClient.PublishEventsWithHttpMessagesAsync(eventGridHost, listEvents);
-                response.Response.EnsureSuccessStatusCode();
+            {
+                await eventGridClient.SendEventsAsync(listEvents);
             }
-            catch (Exception ex) when (ex is ArgumentNullException ||
-                                ex is InvalidOperationException ||
-                                ex is HttpRequestException)
+            catch (Exception ex)
             {
-                throw new EventException("Exception sending event to eventGrid", ex);
-
+                throw new RequestFailedException("Exception sending event to eventGrid", ex);
```

Comment:
The exception that we are catching would already be RequestFailedException. Why are we nesting it inside another RFE?